src/ui/public/kbn_top_nav/bread_crumbs/bread_crumb_url.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * @typedef BreadCrumbUrl {Object}
+ * @property breadcrumb {String} a breadcrumb
+ * @property url {String} a url for the breadcrumb
+ */
+
+/**
+ *
+ * @param {Array.<String>} breadcrumbs An array of breadcrumbs for the given url.
+ * @param {String} url The current url that the breadcrumbs have been generated for
+ * @returns {Array.<BreadCrumbUrl> An array comprised of objects that
+ * will contain both the url for the given breadcrumb, as well as the breadcrumb the url
+ * was generated for.
+ */
+export function getBreadCrumbUrls(breadcrumbs, url) {
```

Comment:
I think this file name should be `bread_crumb_urls.js` (plural) to match up with this function, right?